File: SocialAppFlutter/social_app/lib/Models/message.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:social_app/Models/user.dart';

part 'json_generated_code/message.g.dart';

@JsonSerializable(explicitToJson: true)
class Message {
  final int? id;
  final int fromUser;
  final int? toUser;
  final String content;
  final DateTime datetime;
  final int conversationId;
  final bool isImage;
  final bool isVideo;
  final User? fromUserDetails;
  List<User> usersWhoReacted = [];

  Message(
      {this.id,
      required this.fromUser,
      this.toUser,
      required this.content,
      required this.datetime,
      required this.conversationId,
      required this.isImage,
      required this.isVideo,
      required this.fromUserDetails,
      required this.usersWhoReacted});

  factory Message.fromJson(Map<String, dynamic> json) =>
      _$MessageFromJson(json);

  Map<String, dynamic> toJson() => _$MessageToJson(this);
}

File: APIs/AuthService/AuthService/Models/Message.cs
using System;
using System.Collections.Generic;

namespace AuthService.Models
{
    public partial class Message
    {
        public Message()
        {
            MessageReactions = new HashSet<MessageReaction>();
        }

        public int Id { get; set; }
        public int FromUser { get; set; }
        public int? ToUser { get; set; }
        public string Content { get; set; } = null!;
        public DateTime Datetime { get; set; }
        public int ConversationId { get; set; }
        public bool IsImage { get; set; }
        public bool IsVideo { get; set; }

        public virtual Conversation Conversation { get; set; } = null!;
        public virtual User FromUserNavigation { get; set; } = null!;
        public virtual User? ToUserNavigation { get; set; }
        public virtual ICollection<MessageReaction> MessageReactions { get; set; }
    }
}
